### PR TITLE
Split to crate 

### PR DESCRIPTION
**Pull request by gaetbout**
## Pull Request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [x] Refactoring (no functional changes, no API changes)
- [x] Build-related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

Split the lib into crates

### CODE REVIEW

**Comment by gaetbout** [comment]
@mkaput 
That would be the idea.
Still have to update the makefile, and so on but this is already a first draft

File: quaireaux/data_structures/cairo_project.toml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+[crate_roots]
+data_structures = "src"
+utils = "../utils/src"
+
+tests = "tests"
```
Comment (mkaput) [inline quaireaux/data_structures/cairo_project.toml:5]: I think as for now I would rather go with the convention that exists in `core`: there is `#[cfg(test)]`ed `tests` module within package source

File: cairo_project.toml
[the remark this answers is not in the document]
Comment (gaetbout) [reply cairo_project.toml:1]: Will do!

File: quaireaux/data_structures/cairo_project.toml (hunk 1 above)
Comment (gaetbout) [reply quaireaux/data_structures/cairo_project.toml:5]: Will update 😄

File: cairo_project.toml
[the remark this answers is not in the document]
Comment (gaetbout) [reply cairo_project.toml:1]: Should I put one Scarb.toml in each crate and keep the root one or should I also remove the root one?

[the remark this answers is not in the document]
Comment (mkaput) [reply cairo_project.toml:1]: Root one would make sense if you would like to have a quaireaux package which re-exports subpackages. Like Rust's `crossbeam`: https://docs.rs/crossbeam/latest/crossbeam/

If you don't want to do this, then remove the root package

[the remark this answers is not in the document]
Comment (gaetbout) [reply cairo_project.toml:1]: That'd be a question for @abdelhamidbakhta  then 😅

**Review by abdelhamidbakhta — APPROVED** [review]
lgtm

**Pull request merged**